.github/workflows/preview.yml

```diff
@@ -0,0 +1,31 @@
+name: Preview PR
+
+on:
+  workflow_run:
+    workflows: ["Build and publish"]
+    types: 
+      - completed
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - uses: actions/checkout@v2
+      
+      - name: Find PR
+        if: ${{ github.event.workflow_run.event == 'pull_request' }}
+        run: |
+          PR=$(curl https://api.github.com/search/issues?q=${{ github.event.workflow_run.head_sha }} |
+          grep -Po "(?<=${{ github.event.workflow_run.repository.full_name }}\/pulls\/)\d*" | head -1)
+          echo "PR=$PR" >> $GITHUB_ENV
```

Comment:
I think you should be able to get the PR number just from the ref, e.g. `refs/pull/1234/merge`, see https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#pull_request

Reply:
Unfortunately, that's not exposed to indirect executions as I do here. That'd be nice, though.